octavia_f5/controller/worker/tasks/f5_tasks.py

```
@@ -64,22 +64,33 @@ def execute(self,
             'syncacheThreshold': CONF.networking.syncache_threshold
         }
 
-        device_response = bigip.get(path=f"/mgmt/tm/net/vlan/~Common~{vlan['name']}?expandSubcollections=true")
         # Create vlan if not existing
-        if device_response.status_code == 404:
+        if existing_vlan is None:
             res = bigip.post(path='/mgmt/tm/net/vlan', json=vlan)
             res.raise_for_status()
             return res.json()
 
-        device_vlan = device_response.json()
-        if not vlan.items() <= device_vlan.items():
+        if not vlan.items() <= existing_vlan.items():
             res = bigip.patch(path=f"/mgmt/tm/net/vlan/~Common~{vlan['name']}",
                               json=vlan)
             res.raise_for_status()
             return res.json()
 
         # No Changes needed
-        return device_vlan
+        return existing_vlan
+
+    @decorators.RaisesIControlRestError()
+    def revert(self, network: f5_network_models.Network,
+               bigip: bigip_restclient.BigIPRestClient,
+               existing_vlan, *args, **kwargs):
+        if existing_vlan is not None:
+            LOG.warning(f"Reverting EnsureVLAN: Not deleting VLAN, since it existed before "
+                        f"the task was run: {existing_vlan}")
+            return
+        res = bigip.delete(path=f"/mgmt/tm/net/vlan/~Common~vlan-{network.vlan_id}")
+        if not res.ok:
+            LOG.warning("%s: Failed removing VLAN for vlan_id=%s: %s",
+                        bigip.hostname, network.vlan_id, res.content)
```

Please reraise the error, like this: `res.raise_for_status()`.

---

Good catch, will do